homeassistant/components/shelly/switch.py

```diff
@@ -32,32 +44,56 @@ async def async_setup_entry(
         return
 
     relay_blocks = []
+    assert wrapper.device.blocks
     for block in wrapper.device.blocks:
         if block.type == "relay":
-            appliance_type = wrapper.device.settings["relays"][int(block.channel)].get(
+            app_type = wrapper.device.settings["relays"][int(block.channel)].get(
                 "appliance_type"
             )
-            if not appliance_type or appliance_type.lower() != "light":
-                relay_blocks.append(block)
-                unique_id = (
-                    f'{wrapper.device.shelly["mac"]}-{block.type}_{block.channel}'
-                )
-                await async_remove_shelly_entity(
-                    hass,
-                    "light",
-                    unique_id,
-                )
+            if app_type and app_type.lower() == "light":
```

Comment:
It would be more future proof if you check for allowed app types instead of filtering out bad ones.

Reply:
This is from block device, The device has a general string input for app type, it will accept any string. We have documented that if this is set to "light" we create a light entity:
https://www.home-assistant.io/integrations/shelly/#appliance-type